crates/cdk/src/mint/mod.rs

```diff
@@ -396,82 +330,79 @@ impl Mint {
+    pub fn get_keyset_info(&self, id: &Id) -> Option<MintKeySetInfo> {
+        self.keysets
+            .load()
+            .iter()
+            .filter_map(|keyset| {
+                if keyset.id == *id {
+                    Some(keyset.into())
+                } else {
+                    None
+                }
+            })
+            .next()
+    }
+
     /// Blind Sign
     #[instrument(skip_all)]
     pub async fn blind_sign(
         &self,
         blinded_message: &BlindedMessage,
     ) -> Result<BlindSignature, Error> {
-        let BlindedMessage {
-            amount,
-            blinded_secret,
-            keyset_id,
-            ..
-        } = blinded_message;
-        self.ensure_keyset_loaded(keyset_id).await?;
-
-        let keyset_info = self
-            .localstore
-            .get_keyset_info(keyset_id)
+        self.signatory
+            .blind_sign(vec![blinded_message.to_owned()])
             .await?
-            .ok_or(Error::UnknownKeySet)?;
-
-        // Check that the keyset is active and should be used to sign
-        let active_id = self
-            .localstore
-            .get_active_keyset_id(&keyset_info.unit)
-            .await?
-            .ok_or(Error::InactiveKeyset)?;
-        ensure_cdk!(keyset_info.id.eq(&active_id), Error::InactiveKeyset);
-
-        let keysets = self.keysets.read().await;
-        let keyset = keysets.get(keyset_id).ok_or(Error::UnknownKeySet)?;
-        let key_pair = keyset.keys.get(amount).ok_or(Error::AmountKey)?;
-
-        let c = sign_message(&key_pair.secret_key, blinded_secret)?;
-
-        let blinded_signature = BlindSignature::new(
-            *amount,
-            c,
-            keyset_info.id,
-            &blinded_message.blinded_secret,
-            key_pair.secret_key.clone(),
-        )?;
-
-        Ok(blinded_signature)
+            .pop()
+            .ok_or(Error::Internal)
     }
 
     /// Verify [`Proof`] meets conditions and is signed
     #[instrument(skip_all)]
-    pub async fn verify_proof(&self, proof: &Proof) -> Result<(), Error> {
-        // Check if secret is a nut10 secret with conditions
-        if let Ok(secret) =
-            <&crate::secret::Secret as TryInto<crate::nuts::nut10::Secret>>::try_into(&proof.secret)
-        {
-            // Checks and verifies known secret kinds.
-            // If it is an unknown secret kind it will be treated as a normal secret.
-            // Spending conditions will **not** be check. It is up to the wallet to ensure
-            // only supported secret kinds are used as there is no way for the mint to
-            // enforce only signing supported secrets as they are blinded at
-            // that point.
-            match secret.kind {
-                Kind::P2PK => {
-                    proof.verify_p2pk()?;
-                }
-                Kind::HTLC => {
-                    proof.verify_htlc()?;
+    pub async fn verify_proofs(&self, proofs: &[Proof]) -> Result<(), Error> {
```

Comment:
```suggestion
    pub async fn verify_proofs(&self, proofs: Proofs) -> Result<(), Error> {
```

Since we just call to_owned on this later should we just make this a vec?